#!/usr/bin/env python
PACKAGE = "landing_site_detection"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("leaf_size", double_t, 0, "Voxel filter leaf size, in m", 0.1, 0.0, 1.0)
gen.add("octree_resolution", double_t, 0, "Resolution for octree in m", 0.1, 0.0, 1.0)
gen.add("octree_search_thresh", int_t, 0, "Number of points octree search must find to pass", 40, 1, 200)
gen.add("octree_search_radius", double_t, 0, "Radius of sphere which octree searches around a point", 0.5, 0.01, 2.0)
gen.add("surface_area_thresh", double_t, 0, "Threshold for landing site surface area", 0.4, 0.0, 5.0)
gen.add("density_score_weight", double_t, 0, "Weight of density score", 0.0, 0.0, 1.0)
gen.add("depth_score_weight", double_t, 0, "Weight of depth score", 1.0, 0.0, 1.0)
gen.add("surface_area_score_weight", double_t, 0, "Weight of surface area score", 0.0, 0.0, 1.0)

exit(gen.generate(PACKAGE, "landing_site_detection", "LSD"))
